spec/fixtures/modules/test/lib/puppet/functions/frozen_function.rb

```diff
@@ -1,5 +1,5 @@
 Puppet::Functions.create_function(:frozen_function) do
   def frozen_function(value)
-    value.reverse!
+    value.frozen?
```

Comment:
This kinda changes what is being tested: You will most likely in a real life scenario not check for frozen but rather by accident execute a method on an object that changes it.